ethereum/listener.go

```diff
@@ -129,3 +129,80 @@ func (e *Ethereum) StartListener(
 		}
 	}()
 }
+
+func (e *Ethereum) WalletBalanceMetric(ctx context.Context, logger log.Logger, m *relayer.PromMetrics) {
+	logger = logger.With("metric", "wallet blance", "chain", e.name, "domain", e.domain)
+	queryRate := 30 // seconds
+
+	var err error
+	var client *ethclient.Client
+
+	account := common.HexToAddress(e.minterAddress)
+
+	exponent := big.NewInt(int64(e.MetricsExponent))                                      // ex: 18
+	scaleFactor := new(big.Float).SetInt(new(big.Int).Exp(big.NewInt(10), exponent, nil)) // ex: 10^18
+
+	defer func() {
+		if client != nil {
+			client.Close()
+		}
+	}()
+
+	first := make(chan struct{}, 1)
+	first <- struct{}{}
+	createClient := true
+	for {
+		timer := time.NewTimer(time.Duration(queryRate) * time.Second)
+		select {
+		// don't wait the "queryRate" amount of time if this is the first time running
+		case <-first:
```

Comment:
`case <-first` looks like a duplicate of `case <-timer.C`. Maybe pull the query logic out into its own function then call it after `createClient := true` once, then in time.C too as usual (waiting for the queryRate).

Reply:
Yes good call. That will be dryer. 
In a drafted PR, I have much better handling of these clients. I will clean this up in that PR.